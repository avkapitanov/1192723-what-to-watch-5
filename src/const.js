export const DEFAULT_RATING_FORM_VALUE = 3;
export const MAX_RATING_VALUE_FORM = 5;
export const HOVER_TIME_BEFORE_PLAYING = 2000;
export const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export const DEFAULT_ACTIVE_TAB = FilmTab.OVERVIEW;
export const DEFAULT_GENRE_FILTER_VALUE = `All genres`;
export const INITIAL_FILMS_COUNT = 8;
export const FILMS_COUNT_PER_STEP = 8;
export const MAX_GENRES_IN_FILTER = 9;
export const API_ENDPOINT = `https://5.react.pages.academy/wtw`;
export const REQUEST_TIMEOUT = 5000;
export const HttpResponseCode = {
  UNAUTHORIZED: 401
};
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
export const FilmRating = {
  BAD: {
    VALUE: 3,
    TEXT: `Bad`
  },
  NORMAL: {
    VALUE: 5,
    TEXT: `Normal`
  },
  GOOD: {
    VALUE: 8,
    TEXT: `Good`
  },
  VERY_GOOD: {
    VALUE: 10,
    TEXT: `Very good`
  },
  AWESOME: {
    TEXT: `Awesome`
  }
};

