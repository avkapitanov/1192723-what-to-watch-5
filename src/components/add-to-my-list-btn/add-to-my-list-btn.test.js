import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";

import {AuthorizationStatus} from "../../const";
import AddToMyListBtn from "./add-to-my-list-btn";

const mockStore = configureStore([]);

describe(`AddToMyListBtn component render correctly`, () => {
  it(`With unauthorized user`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter history={browserHistory}>
            <AddToMyListBtn
              filmId={1}
              isFavorite={false}
              onAddToMyListBtnClick={() => {}}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With authorized user`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: AuthorizationStatus.AUTH,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter history={browserHistory}>
            <AddToMyListBtn
              filmId={1}
              isFavorite={false}
              handleMyListBtnClick={() => {}}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
