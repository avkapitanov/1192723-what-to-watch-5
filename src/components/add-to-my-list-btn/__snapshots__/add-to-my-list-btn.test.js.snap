// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToMyListBtn component render correctly With authorized user 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`AddToMyListBtn component render correctly With unauthorized user 1`] = `null`;
